pkg/network/ebpf/c/runtime/usm.c

```diff
@@ -585,11 +585,18 @@ static __always_inline int do_sys_open_helper_exit(struct pt_regs *ctx) {
         return 0;
     }
 
-    // Detect whether the file being opened is a shared library
+// Detect whether the file being opened is a shared library
+// Libraries:
+//    libssl.so -> ssl.so
+// libcrypto.so -> pto.so
+// libgnutls.so -> tls.so
     bool is_shared_library = false;
 #pragma unroll
-    for (int i = 0; i < LIB_PATH_MAX_SIZE - SO_SUFFIX_SIZE; i++) {
-        if (path->buf[i] == '.' && path->buf[i + 1] == 's' && path->buf[i + 2] == 'o') {
+    for (int i = 0; i < LIB_PATH_MAX_SIZE - (SO_SUFFIX_SIZE+3); i++) {
+        if ((path->buf[i] == 's' || path->buf[i] == 'p' || path->buf[i] == 't') &&
+            (path->buf[i+1] == 's' || path->buf[i+1] == 't' || path->buf[i+1] == 'l') &&
+            (path->buf[i+2] == 'l' || path->buf[i+2] == 'o' || path->buf[i+2] == 's') &&
+            path->buf[i+3] == '.' && path->buf[i+4] == 's' && path->buf[i+5] == 'o') {
```

Comment:
allowing permutations like `sts.so`, `ttl.so` and more to pass the validation, please be strict with what we're checking

Reply:
done